### PR TITLE
Travis with codecov 

### PR DESCRIPTION
**Pull request by mattclifford1**
.travis.yml added for travis CI
fatf/tests/test_script.py added containing some dummy tests for pytest to run
inside .travis.yml code coverage of test_script.py sent to codecov

### CODE REVIEW

**Comment by So-Cool** [comment]
[As a *developer* I want the *code* to be continuously *tested*](https://trello.com/c/XytYPFty/8-as-a-developer-i-want-the-code-to-be-continuously-tested)

**Comment by So-Cool** [comment]
[As a *developer* I want to *review* all the opened *pull requests* on GitHub to merge them and get the setup finished](https://trello.com/c/2100X0H5/55-as-a-developer-i-want-to-review-all-the-opened-pull-requests-on-github-to-merge-them-and-get-the-setup-finished)

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  - pip install -r requirements.txt
+  - pip install codecov
+  # - pip install pytest
+  - pip install pytest pytest-cov
+
+#virtualenv:
+#  system_site_packages: true
+
+env:
+  - CODECOV_TOKEN="3b343171-4511-44bd-a4d6-dd7d64aef700"  #get token from https://codecov.io/ --> add new repository
```
Comment (alexhepburn) [inline .travis.yml]: Will need new token when we setup the private repo.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  - "2.7"
+  - "3.6"
+  # - "3.7"  fails build 'Unable to download 3.7 archive. The archive may not exist.''
+
+notifications:
+  email: never
+
+install:
+  - pip install -r requirements.txt
+  - pip install codecov
```
Comment (alexhepburn) [inline .travis.yml]: Maybe combine all these pip installs into one line?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  # - pip install pytest
+  - pip install pytest pytest-cov
+
+#virtualenv:
+#  system_site_packages: true
```
Comment (alexhepburn) [inline .travis.yml]: Do we need the commented out code here?

File: fatf/tests/test_script.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+# def test_fail():
+# 	assert times(1, 2) == 3
+
+def main():
+	test_pass()
+
+#test coverage
+if __name__ == '__main__':
+	main()
```
Comment (alexhepburn) [inline fatf/tests/test_script.py]: Add newline to end of file.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+env:
+  - CODECOV_TOKEN="3b343171-4511-44bd-a4d6-dd7d64aef700"  #get token from https://codecov.io/ --> add new repository
+
+script:
+  - py.test --cov-report xml 
+  - coverage run fatf/tests/test_script.py
```
Comment (alexhepburn) [inline .travis.yml]: Will this only run coverage tests on the tests in test_script.py and not the ones in the subdirectories like *fatf/metrics/tests*. Also is this line necessary as the previous like will produce a coverage report in the local directory.

File: fatf/tests/test_script.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+def times(x, y):
```
Comment (alexhepburn) [inline fatf/tests/test_script.py]: Add a top of file descriptor (see google python style guide). Also potentially explain how the functions you want to be tested have to begin with 'test_' in order for pytest to notice them, and the same with file names.

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,31 @@
+sudo: false    #container
```
Comment (alexhepburn) [inline .travis.yml]: Maybe add a comment block to the top of the file describing what it is and a link to the travis configuration guide online?

File: .travis.yml (hunk 3 above)
Comment (mattclifford1) [reply .travis.yml]: sorry accidently left this in

File: .travis.yml (hunk 5 above)
Comment (mattclifford1) [reply .travis.yml]: I'll try change it to run all files in directory, perhaps using wildcard? something like `coverage run fatf/metrics/tests/*.py`

the coverage line is to set up with codecov.io to see the coverage report in a nice format there. I didnt know pytest did coverage reports, but i imagine that would be hard to access if it saves report in local directory when doing the multiple python CI builds? So guess this is why coverage run ... makes this a bit easier?

File: .travis.yml (hunk 2 above)
Comment (mattclifford1) [reply .travis.yml]: or should i put all inside requirments.txt? and have two different requiremnts - one for development and one for the actual package?

**Comment by mattclifford1** [comment]
Made some of the changes that alex suggested. also added linting to CI build

**Pull request closed without merging**